### PR TITLE
Issue #GEN-132: Added module to expose activities as CSV

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

File: web/modules/custom/genlyd_carto/src/Controller/ExportController.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    $ids = $storage->getQuery()
+      ->condition('type', 'activity')
+      ->condition('status', 1)
+      ->execute();
+    $activities = $storage->loadMultiple($ids);
+
+    $response = new CsvResponse();
+
+    // Set headers in the CSV file.
+    $response->setHeaderRow(['Title', 'Image', 'Address', 'Postel', 'Area', 'Encoded address', 'URL']);
```
Comment (tuj) [inline web/modules/custom/genlyd_carto/src/Controller/ExportController.php]: PostalCode?

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
+<?php
+
```
Comment (tuj) [inline web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: @file?

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+namespace Drupal\genlyd_carto\Render;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+/**
+ *
```
Comment (tuj) [inline web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: @TODO

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+   * @throws \InvalidArgumentException When the HTTP status code is not valid
+   */
+  public function __construct($content = array(), $status = 200, $headers = array()) {
+    $this->headers = new ResponseHeaderBag($headers);
+    $this->setStatusCode($status);
+    $this->setProtocolVersion('1.0');
+    $this->headers->set('Content-Type', 'application/CSV');
+
+    if (!empty($content)) {
+      // @TODO: check format for the content parameter.
```
Comment (tuj) [inline web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: @TODO?

File: web/modules/custom/genlyd_carto/src/Controller/ExportController.php [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    foreach ($activities as $activity) {
+      // Load image and use image style.
+      $file = File::load($activity->get('field_image')->entity->id());
+      $image_uri = ImageStyle::load('activity_teaser')->buildUrl($file->getFileUri());
```
Comment (tuj) [inline web/modules/custom/genlyd_carto/src/Controller/ExportController.php:39]: CamelCase?

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+  public function setContent($content) {
+    if (null !== $content && !is_array($content)) {
+      throw new \UnexpectedValueException(sprintf('The Response content must be a string or object implementing __toString(), "%s" given.', gettype($content)));
+    }
+
+    $this->content = $content;
+
+    return $this;
+  }
+}
```
Comment (tuj) [inline web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: Newline?

**Review by tuj — CHANGES_REQUESTED** [review]

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php (hunk 3 above)
Comment (tuj) [reply web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: Missing documentation

File: web/modules/custom/genlyd_carto/src/Controller/ExportController.php (hunk 5 above)
Comment (cableman) [reply web/modules/custom/genlyd_carto/src/Controller/ExportController.php:39]: Will not fix!

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php (hunk 2 above)
Comment (cableman) [reply web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: done

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php (hunk 3 above)
Comment (cableman) [reply web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: done

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php (hunk 4 above)
Comment (cableman) [reply web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: done

File: web/modules/custom/genlyd_carto/src/Render/CsvResponse.php (hunk 6 above)
Comment (cableman) [reply web/modules/custom/genlyd_carto/src/Render/CsvResponse.php]: done

File: web/modules/custom/genlyd_carto/src/Controller/ExportController.php (hunk 5 above)
Comment (tuj) [reply web/modules/custom/genlyd_carto/src/Controller/ExportController.php:39]: Nå!

**Review by tuj — APPROVED** [review]

**Pull request merged**